### DISCUSSION TITLE
userInputs not working for me

### DISCUSSION THREAD

**Discussion by paulverbeke** [zxcvbn-ts/zxcvbn#68] [Q&A] [answered]
Hi @MrWook hope you're doing well.
I finally got the time to play around with this library.
I'm trying to implement it in one of my project but I cannot make the userInputs dictionnary working for me.
I'm setting these options:

```javascript
const options3 = {
    translations: zxcvbnts['language-fr'].translations,
    graphs: zxcvbnts['language-common'].adjacencyGraphs,
    dictionary: {
        userInputs: ['test456123@gmail.com'],
        ...zxcvbnts['language-common'].dictionary,
        ...zxcvbnts['language-en'].dictionary,
        ...zxcvbnts['language-fr'].dictionary
    },
}
zxcvbnts.core.ZxcvbnOptions.setOptions(options3)
```

calling with this password:
```javascript
zxcvbnts.core.zxcvbn("test456123@gmail.com")
```

and getting this result with a score of 4 and no suggestions -_-:
```javascript
calcTime: 13
crackTimesDisplay: Object { onlineThrottling100PerHour: "siècles", onlineNoThrottling10PerSecond: "siècles", offlineSlowHashing1e4PerSecond: "siècles", … }
crackTimesSeconds: Object { onlineThrottling100PerHour: 271252803600000000, onlineNoThrottling10PerSecond: 753480010000000, offlineSlowHashing1e4PerSecond: 753480010000, … }
feedback: Object { warning: "", suggestions: [] }
guesses: 7534800100000000
guessesLog10: 15.877071734869821
password: "test456123@gmail.com"
score: 4
```

What am I doing wrong here ?

**Comment by MrWook** [comment, marked as answer]
Hey,

thats because you added the userInputs first, move it to the back and it should work. For some reason the userInput directory was added to the language dictionaries too.
I think that there were a reason but i'm not sure anymore. I will look into it and try to remove them from the language dictionaries. This is pretty confusing.

**Reply by paulverbeke** [reply]
Thanks for this I'll try

**Comment by paulverbeke** [comment]
that works thanks for this !
